### PR TITLE
[#163903861]: Enable Asset Export to File

### PR DESCRIPTION
**Pull request by EmaEvidence**
## What does this PR do?
It enables Assets export to file.

## Description of Task to be completed?
Print asset details to file `export data ` button

## How should this be manually tested?
- Navigate to the asset list page
- Click the Export to File button
- Click on the download link available on the modal

## What are the relevant pivotal tracker stories?
[#163903861](https://www.pivotaltracker.com/story/show/163903861)

## Any background context you want to add?
- Nil

## Important notes
- Nil

## Packages installed
- Nil

## Deployment note
- Nil

## Related PRs branch | PR (branch_name) | (pr_link)
Branch | PR
------ | ------
other_pr_production | [link]()
other_pr_master | [link]()
## Todos
- [x] Raise PR
- [x] Test
- [ ] Documentation
## Screenshots
image
image
image

### CODE REVIEW

File: src/_reducers/assets.reducer.js [hunk 1]
```diff
@@ -216,6 +218,26 @@ export default (state = initialState.assets, action) => {
...
       };
 
+    case EXPORT_ASSETS_SUCCESS:
+      return {
+        ...state,
+        exportAsset: {
+          hasError: false,
+          message: action.payload.Success,
+          data: action.payload.file
+          // type: action.type
```
Comment (andela-fojuri) [inline src/_reducers/assets.reducer.js]: Kindly remove commented line

File: src/components/AssetsComponent.jsx [hunk 2]
```diff
@@ -93,23 +94,26 @@ export default class AssetsComponent extends Component {
...
+        />
+        <Filter
+          activePage={this.props.activePage}
+          limit={this.state.limit}
+          filterData={this.props.filterData}
+          selected={this.props.selected}
+          filterSelection={this.props.filterSelection}
+          filterAction={this.props.getAssetsAction}
+          disabled={this.props.isLoading}
+        />
```
Comment (andela-fojuri) [inline src/components/AssetsComponent.jsx]: Please destructure all state and props objects.

Comment (EmaEvidence) [reply src/components/AssetsComponent.jsx]: Removed, thanks

File: src/_reducers/assets.reducer.js (hunk 1 above)
Comment (EmaEvidence) [reply src/_reducers/assets.reducer.js]: Removed, thanks

**Comment by andela-fojuri** [comment]
The Export button should be next to the filters button. Kindly confirm from @MadgeKinyanjui

**Comment by MadgeKinyanjui** [comment]
Yes @andela-fojuri it should be next to the filter.

File: src/__test__/actions/assets.action.test.js [hunk 3]
```diff
@@ -156,4 +157,24 @@ describe('Asset Types action tests', () => {
...
+    const testurl = 'export-assets/';
+    mock.onGet(testurl).reply(200, {});
+    return store.dispatch(exportAssetsAction(testurl)).then(() => {
+      expect(store.getActions()).toContainEqual({
+        type: 'EXPORT_ASSETS_SUCCESS'
+      });
+    });
+  });
+
+  // xit('should dispatch EXPORT_ASSETS_FAILURE, when exportAssetsAction is called', () => {
```
Comment (andela-fojuri) [inline src/__test__/actions/assets.action.test.js]: This test is commented. Kindly address it.

**Comment by andela-fojuri** [comment]
It will be nice to close the modal once the download has started.

**Pull request closed without merging**